clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2416,11 +2416,18 @@ protected long measureLagWithCallToPubSub(
       return Long.MAX_VALUE;
     }
     TopicManager tm = getTopicManager(pubSubServerName);
-    long endOffset = tm.getLatestOffsetCached(topic, partition) - 1;
+    long endOffset = tm.getLatestOffsetCached(topic, partition);
```

Comment:
what if we returned lag `0` if endOffset is `0`?

Reply:
So... I guess the rationale is that if the end offset is zero, then it's an empty partition, and therefore there must be zero lag. We can do that if you think it makes the code clearer. Although the rest of the function would still be the same right? (i.e. we still the -1 later on, etc.)

Reply:
I adopted your suggestion in the latest commit, and also added a unit test.